client/dom/table.ts

```diff
@@ -97,6 +97,51 @@ inputName=string
 	if the same name is always used, multiple tables created in one page will conflict in row selection
 */
 
+export type Cell = {
+	url?: string
+	value?: string | number | undefined
+	color?: string
+	html?: string
+	__td?: any
+	values?: Cell[]
+	disabled?: boolean
+}
+
+export type Column = {
+	label: string
+	width?: string
+	editCallback?: ((i: number, cell: Cell) => void) | undefined
+}
+
+export type Button = {
+	text: string
+	callback: (idxs: number[], button: any) => void //called when the button is clicked
+	disabled?: (index: number) => boolean
+	button: any
+	onChange?: (idx: number[], button: any) => void //Called when selecting rows, it would update the button text
+	class?: string
+}
+
+export type TableArgs = {
+	columns: Column[]
+	rows: Cell[][]
+	div: any
+	columnButtons?: Button[]
+	buttons?: Button[]
+	noButtonCallback?: (i: number, node: any) => void
+	singleMode?: boolean
+	showLines?: boolean
+	striped?: boolean
+	showHeader?: boolean
+	maxWidth?: string
+	maxHeight?: string
+	selectedRows: number[]
```

Comment:
please move comments at the beginning over here to explain purpose of each property